modules/internal/ChapelArray.chpl

```diff
@@ -2136,50 +2142,28 @@ module ChapelArray {
     proc this(args ...rank) where _validRankChangeArgs(args, _value.dom.idxType) {
       if boundsChecking then
         checkRankChange(args);
-      var newD = _dom((...args));
-      var ranges = _getRankChangeRanges(newD.dims());
-      //
-      // TODO: Currently, the domain created to represent the
-      // rank-change domain is non-distributed.  Ultimately, we need
-      // to create a domain view class that supports a rank-change
-      // view on a higher-dimensional domain as in the original array
-      // view attempt.
-      //
-      pragma "no auto destroy" var d = {(...ranges)};
-      d._value._free_when_no_arrs = true;
 
-      //
-      // Compute which dimensions are collapsed and what the index
-      // (idx) is in the event that it is.  These will be stored in
-      // the array view to convert from lower-D indices to higher-.
-      //
-      var collapsedDim: rank*bool;
-      var idx: rank*idxType;
-
-      for param i in 1..rank {
-        if (isRange(args(i))) {
-          collapsedDim(i) = false;
-        } else {
-          collapsedDim(i) = true;
-          idx(i) = args(i);
-        }
-      }
+      pragma "no auto destroy"
+      const rcdom = this.domain[(...args)];
+      rcdom._value._free_when_no_arrs = true;
```

Comment:
It seems likely that removing pragma "no auto destroy" on rcdom and removing the _free_when_no_arrs would simplify things and still be correct. The rcdom will be destroyed at the end of this fn, but because add_arr was called, it will end up not being destroyed yet and will set _free_when_no_arrs = true.